src/components/DeckMechanisms/handleCardEffect.js

```diff
@@ -205,13 +207,15 @@ const handleCardEffect = (state, card, mode) => {
 
     // Collector Mirz
     case 'N8': {
-      state.deck.push(getCollectorMirzToken(card.level))
+      state.deck.push(getCollectorMirzToken(state.deck, card.level))
       break
     }
 
     // First Mutineer
     case 'N12': {
-      const nonPirates = state.hand.filter(getNonPirates(state))
+      const nonPirates = state.hand.filter(card =>
+        isNonPiratesInHand(state, card)
+      )
```

Comment:
I would rename `isNonPiratesInHand` into `isNotPirate`, since the function no longer deals with the hand. I would also have it defined like this:

```js
const isNotPirate = state => card => { … }
```
```js
const nonPirates = state.hand.filter(isNotPirate(state))
```